app/controlplane/plugins/core/webhook/v1/webhook.go

```diff
@@ -0,0 +1,327 @@
+	var attachReq attachmentRequest
+	if err := sdk.FromConfig(req.Payload, &attachReq); err != nil {
+		i.Logger.Errorw("failed to parse attachment payload", "error", err)
+		return nil, fmt.Errorf("invalid attachment request: %w", err)
+	}
+
+	// Set default values if not provided
+	sendAttestation := true
+	if attachReq.SendAttestation != nil {
+		sendAttestation = *attachReq.SendAttestation
+	}
+
+	sendSBOM := false
+	if attachReq.SendSBOM != nil {
+		sendSBOM = *attachReq.SendSBOM
+	}
+
+	// Store the settings in the attachment state
+	rawConfig, err := sdk.ToConfig(&attachmentState{
+		SendAttestation: sendAttestation,
+		SendSBOM:        sendSBOM,
+	})
+	if err != nil {
+		i.Logger.Errorw("failed to marshal attachment state", "error", err)
+		return nil, fmt.Errorf("marshalling attachment state: %w", err)
+	}
+
+	return &sdk.AttachmentResponse{
+		Configuration: rawConfig,
+	}, nil
+}
+
+// Execute is called when an attestation or SBOM is received
+func (i *Integration) Execute(ctx context.Context, req *sdk.ExecutionRequest) error {
+	// Extract the webhook URL from credentials
+	if req.RegistrationInfo.Credentials == nil || req.RegistrationInfo.Credentials.URL == "" {
+		i.Logger.Error("missing webhook URL in credentials")
+		return errors.New("missing webhook URL in credentials")
+	}
+	webhookURL := req.RegistrationInfo.Credentials.URL
+
+	// Extract the settings from the attachment state
+	var attachState attachmentState
+	if err := sdk.FromConfig(req.AttachmentInfo.Configuration, &attachState); err != nil {
+		i.Logger.Errorw("invalid attachment state", "error", err)
+		return fmt.Errorf("invalid attachment state: %w", err)
+	}
+
+	// Send attestation if enabled and present
+	if attachState.SendAttestation && req.Input.Attestation != nil {
+		statementJSON, err := json.Marshal(req.Input.Attestation)
+		if err != nil {
+			i.Logger.Errorw("failed to marshal attestation", "error", err)
+			return fmt.Errorf("marshalling attestation: %w", err)
+		}
+		if err := i.sendWebhook(ctx, webhookURL, "ATTESTATION", statementJSON, req.ChainloopMetadata); err != nil {
+			i.Logger.Errorw("failed to send attestation webhook", "error", err)
+			return err
+		}
+	}
+
+	// Send SBOM if enabled and present
+	if attachState.SendSBOM {
+		for _, material := range req.Input.Materials {
+			// Ensure material type is either SBOM_CYCLONEDX_JSON or SBOM_SPDX_JSON
+			if material.Type != schemaapi.CraftingSchema_Material_MaterialType_name[int32(schemaapi.CraftingSchema_Material_SBOM_CYCLONEDX_JSON)] &&
+				material.Type != schemaapi.CraftingSchema_Material_MaterialType_name[int32(schemaapi.CraftingSchema_Material_SBOM_SPDX_JSON)] {
+				i.Logger.Warnw("unsupported material type, skipping", "type", material.Type)
+				continue
+			}
+			// Validate material content
+			if len(material.Content) == 0 {
+				i.Logger.Warnw("encountered SBOM with empty content, skipping", "type", material.Type)
+				continue
+			}
+
+			materialJSON, err := json.Marshal(material)
```

Comment:
it might not be a good idea to json marshal always the content, can we just sent the array of bytes?